pkg/cloud/services/loadbalancer/loadbalancer.go

```diff
@@ -318,48 +318,80 @@ func (s *Service) ReconcileLoadBalancerMember(openStackCluster *infrav1.OpenStac
 	return nil
 }
 
+// DeleteLoadBalancer deletes a load balancer identified by its name and all child objects (cascade).
 func (s *Service) DeleteLoadBalancer(openStackCluster *infrav1.OpenStackCluster, clusterName string) error {
 	loadBalancerName := getLoadBalancerName(clusterName)
 	lb, err := s.checkIfLbExists(loadBalancerName)
 	if err != nil {
 		return err
 	}
+	return s.deleteLoadBalancer(openStackCluster, lb)
+}
 
+// deleteLoadBalancer deletes the load balancer and all child objects (cascade).
+func (s *Service) deleteLoadBalancer(openStackCluster *infrav1.OpenStackCluster, lb *loadbalancers.LoadBalancer) error {
 	if lb == nil {
 		return nil
 	}
 
 	if lb.VipPortID != "" {
 		fip, err := s.networkingService.GetFloatingIPByPortID(lb.VipPortID)
 		if err != nil {
-			return err
+			return errors.Wrapf(err, "get floating IP by port ID %s", lb.VipPortID)
 		}
 
 		if fip != nil && fip.FloatingIP != "" {
 			if err = s.networkingService.DisassociateFloatingIP(openStackCluster, fip.FloatingIP); err != nil {
-				return err
+				return errors.Wrapf(err, "dissassociate floating IP %s", fip.FloatingIP)
 			}
 			if err = s.networkingService.DeleteFloatingIP(openStackCluster, fip.FloatingIP); err != nil {
-				return err
+				return errors.Wrapf(err, "delete floating IP %s", fip.FloatingIP)
 			}
 		}
 	}
 
 	deleteOpts := loadbalancers.DeleteOpts{
 		Cascade: true,
 	}
-	s.logger.Info("Deleting load balancer", "name", loadBalancerName, "cascade", deleteOpts.Cascade)
+	s.logger.Info("Deleting load balancer", "id", lb.ID, "name", lb.Name, "cascade", deleteOpts.Cascade)
 	mc := metrics.NewMetricPrometheusContext("loadbalancer", "delete")
-	err = loadbalancers.Delete(s.loadbalancerClient, lb.ID, deleteOpts).ExtractErr()
+	err := loadbalancers.Delete(s.loadbalancerClient, lb.ID, deleteOpts).ExtractErr()
 	if mc.ObserveRequest(err) != nil {
 		record.Warnf(openStackCluster, "FailedDeleteLoadBalancer", "Failed to delete load balancer %s with id %s: %v", lb.Name, lb.ID, err)
-		return err
+		return errors.Wrapf(err, "delete loadbalancer by ID %s", lb.ID)
 	}
 
 	record.Eventf(openStackCluster, "SuccessfulDeleteLoadBalancer", "Deleted load balancer %s with id %s", lb.Name, lb.ID)
 	return nil
 }
 
+// DeleteLoadBalancersByNetworkID deletes all remaining load balancers in the network
+// identified by openStackCluster.Status.Network.ID.
+func (s *Service) DeleteLoadBalancersByNetworkID(openStackCluster *infrav1.OpenStackCluster) error {
+	if openStackCluster.Status.Network == nil || openStackCluster.Status.Network.ID == "" {
+		return errors.Errorf("expected status.network.id for OpenStackCluster %s/%s", openStackCluster.Namespace, openStackCluster.Name)
+	}
+	lbList, err := loadbalancers.List(s.loadbalancerClient, loadbalancers.ListOpts{
+		VipNetworkID: openStackCluster.Status.Network.ID,
```

Comment:
Please let me know if we should get the network id similar to https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/221d55b6112e92b90f2f946acaeaffc37baecfd7/pkg/cloud/services/networking/network.go#L148-L149 or do it like I proposed.

Reply:
On the contrary, I think we should update DeleteNetwork to delete by ID.